PROTOCOL.md

```diff
@@ -0,0 +1,75 @@
+# The XMTP Protocol
+
+## Intro
+
+The protocol allows `clients`, representing identities associated with blockchain accounts (further 'accounts'), to communicate securely with each other. The communications are represented as `conversations` between two or more clients. A conversation consists of a sequence of `messages` from a `sender` to the other `participants` of the conversation, the `recipients`.
+
+The communication is mediated by a network of `nodes`. The network is responsible for ensuring that messages are delivered between clients. Since clients are not necessarily connected to the network at the same time, the network must be able to store messages while they are in transit.
+
+The protocol is layered to satisfy various operational and security requirements. For example, the clients are responsible for all the encryption and authentication, and key management. The nodes generally see only encrypted content with no indication of identities involved.
+
+## Network Layer
+
+Messages are opaque to the node (usually encrypted), i.e. nodes do not read/interpret the contents of the messages. Unencrypted message headers are readable and can be processed by the nodes.
```

Comment:
Same here with "usually" -- when are messages unencrypted? Seems like it still works without it:

Messages are opaque to the node (encrypted), i.e. nodes do not read/interpret the contents of the messages. Unencrypted message headers are readable and can be processed by the nodes.

Reply:
I'm hedging because few paragraphs below I'm proposing the nodes to actually parse and validate contact bundles. Maybe that's not really necessary. Maybe I should drop that for now, @neekolas was pushing against that as well. But even then I'm a bit iffy about a categoric statement that nodes will not interpret the payload under any circumstances.

Reply:
Then again, given the terminology I'm pushing here where `message` means payload with user generated content, then those would always be encrypted. I'll review and make sure to use `payload` for arbitrary envelope content and `message` for payload with user generated content.

Reply:
>I'm hedging because few paragraphs below I'm proposing the nodes to actually parse and validate contact bundles.

I think the piece that might be confusing is does that go against the next part of the sentence, "i.e. nodes do not read/interpret the contents of the messages"? Maybe we can untangle this by listing out what is encrypted and what is not -- almost like a series of facts instead of a general statement? Idk if that introduces too much complexity early on though and I def trust your gut here! I don't want to pigeon-hole us.